content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+        extraVolumes:
+          - name: policies
+            hostPath: /etc/kubernetes/policies
+            mountPath: /etc/kubernetes/policies
+            readOnly: true
+            pathType: "DirectoryOrCreate"
+  # mount the local file on the control plane
+  extraMounts:
+  - hostPath: ./pod-security.yaml
+    containerPath: /etc/kubernetes/policies/pod-security.yaml
+    readOnly: true
+EOF
+```
+
+```shell
+kind create cluster --image kindest/node:v1.23.0 --config kind-config.yaml
+```
+
+Let's look at the default namespace.
+
+```shell
+kubectl describe namespace default
+```
+
+The output is similar to this:
+
+```shell
+Name:         default
+Labels:       kubernetes.io/metadata.name=default
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Let's create a new namespace and see if the labels apply there.
+
+```shell
+kubectl create namespace test-defaults
+kubectl describe namespace test-defaults
+```
+
+Same.
+
+```shell
+Name:         test-defaults
+Labels:       kubernetes.io/metadata.name=test-defaults
+Annotations:  <none>
+Status:       Active
+
+No resource quota.
+
+No LimitRange resource.
+```
+
+Can we deploy a workload that is privileged?
+
+```shell
+cat <<EOF | kubectl -n test-defaults apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+Hmm... yep. Our default `warn` level is working at least.
+
+```shell
+Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+pod/nginx-privileged created
+```
+
+Let's delete the pod with `kubectl -n test-defaults delete pod/nginx-privileged`.
+
+Is my config even working?
+
+```shell
+# shell into the control plane
+docker exec -it kind-control-plane bash
+
+# cat out the file we mounted
+cat /etc/kubernetes/policies/pod-security.yaml
+
+# check the api server logs
+cat /var/log/containers/kube-apiserver*.log 
+
+# check the api server config
+cat /etc/kubernetes/manifests/kube-apiserver.yaml 
+```
+
+**UPDATE:** The baseline policy permits `allowPrivilegeEscalation`. While I cannot see the Pod Security default levels of enforcement, they are there. Let's try to provide a manifest that violates the baseline by requesting hostNetwork access.
+
+```shell
+cat <<EOF | kubectl -n test-defaults apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+  hostNetwork: true
+EOF
+```
```

Comment:
aside: In docs, but not blog articles, we usually use the `codenew` shortcode and tell people to apply a manifest by its URL.